src/jit/indirectcalltransformer.cpp

```diff
@@ -428,19 +428,19 @@ class IndirectCallTransformer
         //
         void AddHiddenArgument(GenTreeCall* fatCall, GenTree* hiddenArgument)
```

Comment:
It can also be appended to the list depending on `USER_ARGS_COME_LAST`. I believe that the function would be clear enough and did not require a comment, if it weren't for the "manual" linked list manipulation that happens in the ret buffer arg case. So I cleaned up the "manual" linked list manipulation instead of adding a comment. Anyway it was messed up - instead of simply inserting a new use for the hidden argument it was also recreating the use of the ret buffer arg.